### PR TITLE
First pass at fixing chemistry costs to use QECGatesCost

### PR DESCRIPTION
**Pull request by fdmalone**

### CODE REVIEW

File: qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py [hunk 1]
```diff
@@ -13,36 +13,25 @@
...
-    return bloq
-
-
 def test_second_order_suzuki_costs():
     length = 8
     u = 4
     dt = 0.1
     unitary = build_plaq_unitary_second_order_suzuki(length, u, dt)
-    _, sigma = unitary.call_graph(generalizer=catch_rotations)
+    # _, sigma = unitary.call_graph(generalizer=catch_rotations)
```
Comment (anurudhp) [inline qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py:29]:
```suggestion
```

**Comment by mpharrigan** [comment]
can you add a minimal PR description

File: qualtran/bloqs/chemistry/pbc/first_quantization/prepare_t_test.py [hunk 2]
```diff
@@ -33,15 +33,19 @@ def test_prepare_kinetic_t_counts():
...
-    _, counts = uni.call_graph()
-    qual_cost = counts[Toffoli()]
+    counts = get_cost_value(uni, QECGatesCost())
+    cost_dict = counts.total_t_and_ccz_count()
+    qual_cost = cost_dict['n_ccz']
     uni = UniformSuperpostionIJFirstQuantization(eta, num_bits_rot_aa=b_r).adjoint()
-    _, counts = uni.call_graph()
-    qual_cost += counts[Toffoli()]
+    counts = get_cost_value(uni, QECGatesCost())
+    cost_dict = counts.total_t_and_ccz_count()
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/pbc/first_quantization/prepare_t_test.py:41]: you can use `counts.total_toffoli_only` now, which saves a line and will complain if there are also T's and rotations

**Review by mpharrigan — COMMENTED** [review]
looks like a step in the right direction! can you answer @anurudhp 's question

File: qualtran/bloqs/chemistry/trotter/hubbard/interaction_test.py [hunk 3]
```diff
@@ -11,12 +11,31 @@
...
+        if isinstance(bloq.angle, float) and abs(bloq.angle) < 1e-12:
+            return ArbitraryClifford(1)
+        else:
+            return attrs.evolve(bloq, angle=PHI)
+    if isinstance(bloq, ZPowGate):
+        if isinstance(bloq.exponent, float) and abs(bloq.exponent) < 1e-12:
+            return ArbitraryClifford(1)
+        else:
+            return attrs.evolve(bloq, exponent=PHI, global_shift=0)
+    return bloq
```
[the remark this answers is not in the document]
Comment (fdmalone) [reply qualtran/bloqs/chemistry/trotter/hubbard/interaction_test.py:36]: Good question, we don't QECGatesCost is more correct, the rotations which were being counted were really clifford (exponent = 2), I'll update the test so it's a non-trivial rotation so the bloq counting logic is correct.

File: qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py]: Done.

File: qualtran/bloqs/chemistry/pbc/first_quantization/prepare_t_test.py (hunk 2 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/pbc/first_quantization/prepare_t_test.py:41]: Done./

Comment (fdmalone) [reply qualtran/bloqs/chemistry/pbc/first_quantization/prepare_t_test.py]: Done

**Review by mpharrigan — APPROVED** [review]
sweeeeeeeeet

**Comment by mpharrigan** [comment]
it wants you to re-render the hubbard notebook (presumably because of the changed angle)

File: qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py [hunk 4]
```diff
@@ -13,36 +13,23 @@
...
+    dt = 0.1234
     unitary = build_plaq_unitary_second_order_suzuki(length, u, dt)
-    _, sigma = unitary.call_graph(generalizer=catch_rotations)
+    costs = get_cost_value(unitary, QECGatesCost())
     # there are 3 hopping unitaries contributing 8 Ts from from the F gate
-    assert sigma[TGate()] == (3 * length**2 // 2) * 8
+    assert costs.total_t_count(ts_per_rotation=0) == (3 * length**2 // 2) * 8
     # 3 hopping unitaries and 2 interaction unitaries
-    assert sigma[Rz(PHI)] == (3 * length**2 + 2 * length**2)
+    assert costs.rotation == (3 * length**2 + 2 * length**2), 3 * length**2 + 2 * length**2
```
Comment (anurudhp) [inline qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py:32]: is the last part a debug message?

**Review by anurudhp — APPROVED** [review]

File: qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py (hunk 4 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/trotter/hubbard/trotter_step_test.py:32]: whoops

**Pull request merged**